clients/cobol-lsp-vscode-extension/src/extension.ts

```diff
@@ -154,85 +97,58 @@ export async function activate(
   telemetryEvent(
     "analysis-mode",
     ["bootstrap", "analysis-mode"],
-    `COBOL LS is being used in ${vscode.workspace.getConfiguration().get(ANALYSIS_MODE) as string} mode`,
+    `COBOL LS is being used in ${SettingsService.getAnalysisMode()} mode`,
   );
 
-  // Register Commands
-  registerCommands(context);
-
-  registerCodeActions(context);
+  await createExtensionFolder(context);
 
-  context.subscriptions.push(
-    vscode.languages.registerCompletionItemProvider(
-      { language: LANGUAGE_ID },
-      new SnippetCompletionProvider(context),
-    ),
-  );
+  initSmartTab(context);
 
-  context.subscriptions.push(
-    vscode.languages.registerCompletionItemProvider(
-      [LANGUAGE_ID, EXP_LANGUAGE_ID, HP_LANGUAGE_ID],
-      new CopybooksCompletionProvider(),
-    ),
-  );
+  let externalApis: ExternalAPIsService | undefined = undefined;
 
-  context.subscriptions.push(
-    vscode.languages.registerCompletionItemProvider(
-      [LANGUAGE_ID, EXP_LANGUAGE_ID, HP_LANGUAGE_ID],
-      new SubroutinesCompletionsProvider(),
-    ),
-  );
+  const languageClientService = await initializeLanguageClientService(context, {
+    executeCommand: (command, args, next) => {
+      if (command == "missing copybook") {
+        externalApis?.clearProfiles();
+      }
+      next(command, args);
+    },
+  });
 
-  context.subscriptions.push(
-    vscode.workspace.onDidChangeTextDocument((event) =>
-      analysisService.invalidate(event.document.uri.toString(), false),
-    ),
+  externalApis = await initializeExternalAPIs(
+    context.globalStorageUri,
+    async () => {
+      if (languageClientService) {
+        await languageClientService.invalidateConfiguration();
+      }
+    },
```

Comment:
This is once again guaranteed to exist, correct?
```suggestion
    languageClientService.invalidateConfiguration,
```